File: 1995-count-special-quadruplets/1995-count-special-quadruplets.java
import java.util.Arrays;

class Solution {
    public int countQuadruplets(int[] nums) {
        // Arrays.sort(nums);
        int cnt = 0;
        for (int a = 0; a < nums.length; a++) {
            for (int b = a + 1; b < nums.length; b++) {
                for (int c = b + 1; c < nums.length; c++) {
                    for (int d = c + 1; d < nums.length; d++) {
                        int sum = nums[a] + nums[b] + nums[c];
                        if(sum == nums[d]) cnt++;
                        // else if(sum < nums[d]) break;
                    }
                }
            }
        }
        return cnt;
    }
}

File: 1995-count-special-quadruplets/1995-count-special-quadruplets.kt
class Solution {
    fun countQuadruplets(nums: IntArray): Int {

        var cnt: Int = 0

        for (a in 0 until nums.size - 3) {
            for (b in a + 1 until nums.size - 2) {
                for (c in b + 1 until nums.size - 1) {
                    for (d in c + 1 until nums.size) {
                        if(nums[a] + nums[b] + nums[c] == nums[d]) cnt++
                    }
                }
            }
        }
        
        return cnt;
    }
}